### PR TITLE
tests(fix): remove defer on tests

### PR DESCRIPTION
**Pull request by jLopezbarb**
# Proposed changes

Fixes Make explicit call to remove namespace. We have a bug that the defer was not executed on the last tests, leaving a namespace without deleting.

This change has one side effect. Now the tests that fails will not remove the ns that fails in order to look into it

## CLI Quality Reminders 🔧

For both authors and reviewers:

- Scrutinize for potential regressions
- Ensure key automated tests are in place
- Build the CLI and test using the validation steps
- Assess Developer Experience impact (log messages, performances, etc)
- If too broad, consider breaking into smaller PRs
- Adhere to our [code style](https://github.com/okteto/okteto/blob/master/docs/code-style.md) and [code review](https://github.com/okteto/okteto/blob/master/docs/code-review.md) guidelines

### CODE REVIEW

**Review by maroshii — APPROVED** [review]
> Now the tests that fails will not remove the ns that fails in order to look into it

I think this is acceptable

**Comment by andreafalzetti** [comment]
@jLopezbarb I've added the label to run the e2e tests so we can try the change

**Review by andreafalzetti — APPROVED** [review]
Approving, just check the e2e tests if they pass before merging 🙏

**Pull request merged**
